pkg/controllers/capiinstaller/capi_installer_controller.go

```diff
@@ -327,14 +269,20 @@ func (r *CapiInstallerController) SetupWithManager(mgr ctrl.Manager) error {
 			&corev1.ConfigMap{},
 			handler.EnqueueRequestsFromMapFunc(toClusterOperator),
 			builder.WithPredicates(configMapPredicate(r.ManagedNamespace, r.Platform)),
+		).
+		// We reconcile all Deployment changes because we intend to reflect the
+		// status of any created Deployment in the ClusterOperator status.
+		Watches(
+			&appsv1.Deployment{},
+			handler.EnqueueRequestsFromMapFunc(toClusterOperator),
+			builder.WithPredicates(ownedPlatformLabelPredicate(r.ManagedNamespace, r.Platform)),
```

Comment:
I remember when we discussed this we decided to leave this in because of functionality we intend to implement in the future. Honestly I'd leave it out and add it back when we add something which needs it. Until then it's just adding noise.

Reply:
Yes. See my comment above re:
`TODO: Deployments State/Conditions should influence the overall ClusterOperator Status`.

I think ideally we should be looking at doing this considering the upcoming GAing of the operator.

Maybe something we can tackle in tandem with the un-revert of: https://github.com/openshift/cluster-capi-operator/pull/273

In this PR we should at least see if we are introducing anything that would prevent us easily doing that.